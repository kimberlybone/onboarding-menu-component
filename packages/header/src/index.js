export { MainTitle } from './components/MainTitle';
export { Status } from './components/Status';
