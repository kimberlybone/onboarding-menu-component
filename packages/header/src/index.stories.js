import React                              from 'react';
import { storiesOf, addDecorator }        from '@storybook/react';
import styled                             from "styled-components";
import { withKnobs, select, text }				from '@storybook/addon-knobs';
import { MainTitle, Status }              from './index';
storiesOf( 'Header', module )
  .addDecorator(withKnobs)
  .add('MainTitle', () => {
    return <MainTitle></MainTitle>
  })
  .add('Status', () => {
    return <Status></Status>
  })
