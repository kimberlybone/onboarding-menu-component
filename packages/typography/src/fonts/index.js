import WebFont from 'webfontloader';

const WebFontConfig = {
  custom: {
    families: ['GothamNarrowUltra', 'GothamNarrowLight'],
    urls: ['http://www.tightventures.com/fonts/export.css']
  },
  google:{
  	families: ['Open Sans']
  }
};

WebFont.load(WebFontConfig);

export const Fonts = {
	GothamNarrowUltra: 'GothamNarrowUltra',
	GothamNarrowLight: 'GothamNarrowLight',
  OpenSans: 'Open Sans'
};
