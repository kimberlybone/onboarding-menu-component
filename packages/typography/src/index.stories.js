import React                            from 'react';
import { storiesOf, addDecorator }      from '@storybook/react';
import { withKnobs, text, select }      from '@storybook/addon-knobs';
import { Jumbo, 
         SubJumbo, 
         Heading, 
         SubHeading, 
         Body, 
         TitleHighlight }               from './index';
import { Colors }                       from '@narrow/theme';

const primitiveOptions = {
  'h1': 'h1',
  'h2': 'h2',
  'h3': 'h3',
  'h4': 'h4',
  'h5': 'h5',
  'h6': 'h6',
  'p' : 'p',
  'span': 'span'
};

const colorOptions = {
  GrayVeryDark: Colors.GrayVeryDark,
  Blue: Colors.Blue,
  Orange: Colors.Orange,
  Purple: Colors.Purple,
  Green: Colors.Green,
  Red: Colors.Red,
  Yellow: Colors.Yellow,
  GrayDark: Colors.GrayDark,
  Gray: Colors.Gray,
  GrayLight: Colors.GrayLight,
  White: Colors.White
};

const sizeOptions = {
  ExtraLarge: 'extraLarge',
  Large: 'large',
  Medium: 'medium',
  Small: 'small',
  ExtraSmall: 'extraSmall'
};

storiesOf('Typography', module)
  .addDecorator(withKnobs)
  .add('Jumbo', () => {
    const size = select('Size', sizeOptions, 'extraLarge');
    return <Jumbo
    {...{[size]: true}}
    color={select('Color', colorOptions)}
    primitive={select('Primitive', primitiveOptions)}
    >
      {text('Title', 'Jumbo')}
    </Jumbo>;

  })
  .add('Heading', () => {
    const size = select('Size', sizeOptions, 'extraLarge');
    return <Heading
    {...{[size]: true}}
    color={select('Color', colorOptions)}
    primitive={select('Primitive', primitiveOptions)}
    >
      {text('Title', 'Heading')}
    </Heading>;

  })
  .add('Body', () => {
    const size = select('Size', sizeOptions, 'extraLarge');
    return <Body
    {...{[size]: true}}
    color={select('Color', colorOptions)}
    primitive={select('Primitive', primitiveOptions)}
    >
      {text('Title', 'Body')}
    </Body>;

  })
  .add('TitleHighlight', () => {
    const size = select('Size', sizeOptions, 'extraLarge');
    return <TitleHighlight
    {...{[size]: true}}
    color={select('Color', colorOptions)}
    lineColor={select('Line Color', colorOptions)}>
      {text('Title', 'Title')}
    </TitleHighlight>;

  });