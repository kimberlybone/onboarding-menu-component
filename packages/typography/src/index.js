import { Fonts } from './fonts';
import { Jumbo } from './components/Jumbo';
import { Heading } from './components/Heading';
import { Body } from './components/Body';
import { TitleHighlight } from './components/TitleHighlight';
import Text  from './components/Text';

export { Fonts, Jumbo, Heading, Body, Text, TitleHighlight };