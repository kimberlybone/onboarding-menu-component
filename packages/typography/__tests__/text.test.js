import React        from 'react';
import renderer     from 'react-test-renderer';
import { Text }         from '../src';

jest.useFakeTimers();

let testFn = jest.fn();

////////////////Snapshot////////////////
test('Text renders correctly', () => {
  const tree = renderer.create(<Text title={'Text'}/>).toJSON();
  expect(tree).toMatchSnapshot();
});

////////////////Default props////////////////
test('renders default props correctly', () => {
  const testRenderer = renderer.create(<Text 
                                        title={'Text'}/>);

  const testInstance = testRenderer.root;

  expect(testInstance.props.primitive).toBe('span');
});

