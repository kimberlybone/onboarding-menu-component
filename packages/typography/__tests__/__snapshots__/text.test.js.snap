// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text renders correctly 1`] = `
.c0 {
  font-size: px;
  line-height: px;
  margin: 0;
}

<span
  className="c0"
  title="Text"
/>
`;
